### PR TITLE
Recursion docs and examples

### PR DESCRIPTION
**Pull request by sklam**
For #2130

### CODE REVIEW

**Comment by sklam** [comment]
This will need a rebase once #2130 is merged

File: vsdebug.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import sys
```
Comment (sklam) [inline vsdebug.py]: oops, added this accidentally

Comment (sklam) [inline vsdebug.py]: removed now

**Comment by sklam** [comment]
@stuartarchibald , I have fixed and clarified the doc.

**Comment by stuartarchibald** [comment]
Thanks for this document. It does a good job of explaining recursion support and some of the internals involved, the example presented is also good. Merging.

**Pull request merged**